### PR TITLE
Sabrina/data model

### PR DESCRIPTION
**Pull request by sabrina-li**
add room DB model, Dao and repo for interfacing with sqlite
generalizing fruit to product to handle generic shoppe items
when user add/remove from cart, the product data is saved into db

### CODE REVIEW

File: java/app/src/main/java/com/fullstorydev/shoppedemo/data/ProductRepository.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    // You must call this on a non-UI thread or your app will throw an exception. Room ensures
+    // that you're not doing any long running operations on the main thread, blocking the UI.
+    public void insertAll(Product product) {
+        AppDatabase.databaseWriteExecutor.execute(() -> {
+            mProductDao.insertAll(product);
+        });
+    }
+
+    public boolean updateQuantityInCartByOne(Product product,boolean isAdd){
```
Comment (patrick-fs) [inline java/app/src/main/java/com/fullstorydev/shoppedemo/data/ProductRepository.java]: How would you feel about refactoring this into 2 methods: `addProduct(Product product)` and `removeProduct(Product product)`

**Comment by van-fs** [comment]
I saw this PR had a DAO and Database.  At some point, we're going to read products from an external service eventually.  So this PR would allow progress and keep in mind that we'll switch the call for products from a local DB to a RESTful service in the very near future.  To that end, you might consider some internal service layer that allows you to switch from DB to RESTful resource.

File: java/app/src/main/java/com/fullstorydev/shoppedemo/data/ProductRepository.java (hunk 1 above)
Comment (sabrina-li) [reply java/app/src/main/java/com/fullstorydev/shoppedemo/data/ProductRepository.java]: refactored the methods in DAO, use transaction for upsert DB operations and moved them off ui thread with ExecutorService.

**Comment by sabrina-li** [comment]
> I saw this PR had a DAO and Database. At some point, we're going to read products from an external service eventually. So this PR would allow progress and keep in mind that we'll switch the call for products from a local DB to a RESTful service in the very near future. To that end, you might consider some internal service layer that allows you to switch from DB to RESTful resource.

Mainly using DAO and DB for persisting data that user had added in Cart, the list of product will be fetched over REST API.. Fetching every time app launches and MAYBE saved to and synced with DB, or could simply hold in memory depending on the size of the data.

File: java/app/src/main/java/com/fullstorydev/shoppedemo/data/Product.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import androidx.room.Ignore;
+import androidx.room.PrimaryKey;
+
+/*    sample data:
+{
+        "title": "Bananas",
+            "description": "The old reliable.",
+            "price": 2.99,ΩΩΩΩ
+            "image": "banans.jpg",
+            "unit": "lb.", // this is called quantity in json api
```
Comment (patrick-fs) [inline java/app/src/main/java/com/fullstorydev/shoppedemo/data/Product.java]: yep - I realize I misnamed this field. It should be "unit" - I'll fix up.

**Pull request merged**